docs/addons/flagger.md

```diff
@@ -0,0 +1,64 @@
+## Flagger Add-On
+
+[Flagger](https://flagger.app/) is a progressive delivery tool that automates the release process for applications running on Kubernetes. It reduces the risk of introducing a new software version in production by gradually shifting traffic to the new version while measuring metrics and running conformance tests. The Flagger add-on provisions the necessary Helm chart, and namespace to allow support for flagger in an EKS workload.
+
+## Usage
+
+```typescript
+import 'source-map-support/register';
+import * as cdk from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const app = new cdk.App();
+
+const addOn = new blueprints.addons.Flagger();
+
+const blueprint = blueprints.EksBlueprint.builder()
+  .addOns(addOn)
+  .build(app, 'my-stack-name');
+```
+
+## Functionality
+
+1. Creates the `flagger` namespace.
+2. Deploys the `flagger` Helm chart into the cluster.
+3. Supports [standard helm configuration options](./index.md#standard-helm-add-on-configuration-options)
+
+## Configuration Options
+
+- `prometheus`: Pass true or false if you wish to track deployment metrics via prometheus. `True by default`.
+- `crd`: Pass true or false. If true, create Flagger's CRDs (should be enabled for Helm v2 ONLY). If used in Helm v3 it will break the deployment so its `false by default`.
+- `meshProvider`: Pass from the following enum list of meshProviderOptions. `KUBERNETES by default`.
+
+```typescript
+export const enum MeshProviderOptions {
+  KUBERNETES = 'kubernetes',
+  ISTIO = 'istio',
+  LINKERD = 'linkerd',
+  APPMESH = 'appmesh',
+  CONTOUR = 'contour',
+  NGINX = 'nginx',
+  GLOO = 'gloo',
+  SKIPPER = 'skipper',
+  TRAEFIK = 'traefik',
+  OSM = 'osm'
+}
+```
+
+- Example Configuration:
+
+```typescript
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+
+const flagger = new blueprints.addons.FlaggerAddOn({ 
```

Comment:
Why would you provide an example which ignores your own configuration options and leverages values directly? It defeats the purpose.
Please drop the values (that falls under the standard helm add-on configuraton) and provide an example that is using root level properties only (installPrometheus and meshProvider).
With respect to `crd.install`: you mentioned that it is not needed, and moreover if customers pass it as `true` stack will fail? Or is it needed?

Reply:
We should only demonstrate the use of our root level props and provide a reference to the yaml file if they want to pass something else. It does not make sense to show customers how to ignore your properties and go straight to values. If you want to show usage of values, please pick something that does not overlap with root level properties.

With respect to CRD: we are creating an addon so that customer don't have to use helm. It is irrelevant what version of helm customer installed on their laptop. Our lambda layer is using helm3, that is the only thing that matters here.

Reply:
Okay then I would say its best to remove it then if that is the case